opensearch-operator/pkg/reconcilers/cluster.go

```diff
@@ -133,6 +133,11 @@ func (r *ClusterReconciler) Reconcile() (ctrl.Result, error) {
 		})
 		result.CombineErr(err)
 	}
+
+	// Update the cluster health and available nodes in the status
+	result.CombineErr(r.UpdateClusterHealth())
+	result.CombineErr(r.UpdateAvailableNodes())
```

Comment:
I think this should happen as the last point in the reconcile function, after everything else